src/main/java/net/minecraftforge/registries/RegistryObject.java

```diff
@@ -21,323 +17,67 @@
-        if (vanillaRegistry != null)
-        {
-            updateReference(vanillaRegistry);
-            return;
-        }
-
-        this.value = null;
-        this.holder = null;
-    }
-
-    void updateReference(RegisterEvent event)
-    {
-        IForgeRegistry<? extends T> forgeRegistry = event.getForgeRegistry();
-        if (forgeRegistry != null)
-        {
-            updateReference(forgeRegistry);
-            return;
-        }
-
-        Registry<? extends T> vanillaRegistry = event.getVanillaRegistry();
-        if (vanillaRegistry != null)
-            updateReference(vanillaRegistry);
-        else
-            this.value = null;
-    }
-
-    private static boolean registryExists(ResourceLocation registryName)
-    {
-        return RegistryManager.ACTIVE.getRegistry(registryName) != null
-                || BuiltInRegistries.REGISTRY.containsKey(registryName);
-    }
-
-    public ResourceLocation getId()
-    {
-        return this.name;
-    }
-
-    /**
-     * Returns the resource key that points to the registry and name of this registry object.
-     * Nullable only if this RegistryObject is empty and has no name.
-     *
-     * @return the resource key that points to the registry and name of this registry object
-     */
-    @Nullable
-    public ResourceKey<T> getKey()
-    {
-        return this.key;
-    }
-
-    public Stream<T> stream() {
         return isPresent() ? Stream.of(get()) : Stream.of();
     }
-
-    /**
-     * Return {@code true} if there is a mod object present, otherwise {@code false}.
-     *
-     * @return {@code true} if there is a mod object present, otherwise {@code false}
-     */
-    public boolean isPresent() {
-        return this.value != null;
+    
+    private RegistryObject(ResourceKey<T> key)
+    {
+        super(key);
     }
 
-    /**
-     * If a mod object is present, invoke the specified consumer with the object,
-     * otherwise do nothing.
-     *
-     * @param consumer block to be executed if a mod object is present
-     * @throws NullPointerException if mod object is present and {@code consumer} is
-     * null
-     */
     public void ifPresent(Consumer<? super T> consumer) {
         if (isPresent())
             consumer.accept(get());
     }
 
-    /**
-     * If a mod object is present, and the mod object matches the given predicate,
-     * return an {@code RegistryObject} describing the value, otherwise return an
-     * empty {@code RegistryObject}.
-     *
-     * @param predicate a predicate to apply to the mod object, if present
-     * @return an {@code RegistryObject} describing the value of this {@code RegistryObject}
-     * if a mod object is present and the mod object matches the given predicate,
-     * otherwise an empty {@code RegistryObject}
-     * @throws NullPointerException if the predicate is null
-     */
     public RegistryObject<T> filter(Predicate<? super T> predicate) {
```

Comment:
I have staged changed for another PR to swap the test mods and internal code to utilize DH - I was trying to avoid bloating this PR.

Reply:
From my looking at usages of methods though this is the only actual usage of RO specific methods in the code base? Which means it would basically be the only piece of internal code that needs adjustment? (Other than in test mods using it RO as field types.)